geomstats/special_orthogonal_group.py

```diff
@@ -144,114 +164,140 @@ def regularize(self, point):
-                metric = self.left_canonical_metric
-            base_point = self.regularize(base_point)
-
-            jacobian = self.jacobian_translation(
-                                          point=base_point,
-                                          left_or_right=metric.left_or_right)
-            inv_jacobian = gs.linalg.inv(jacobian)
-            tangent_vec_at_id = gs.dot(
-                    tangent_vec,
-                    gs.transpose(inv_jacobian, axes=(0, 2, 1)))
-            tangent_vec_at_id = gs.squeeze(tangent_vec_at_id, axis=1)
-
-            tangent_vec_at_id = self.regularize_tangent_vec_at_identity(
-                                          tangent_vec_at_id,
-                                          metric)
-
-            regularized_tangent_vec = gs.dot(tangent_vec_at_id,
-                                             gs.transpose(jacobian,
-                                                          axes=(0, 2, 1)))
-            regularized_tangent_vec = gs.squeeze(regularized_tangent_vec,
-                                                 axis=1)
-        else:
-            # TODO(nina): is regularization needed in nD?
-            regularized_tangent_vec = tangent_vec
+        if point_representation is None:
+            point_representation = self.point_representation
+
+        if point_representation == 'vector':
+            tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=2)
+            _, vec_dim = tangent_vec.shape
+            if vec_dim == 3:
+                if metric is None:
+                    metric = self.left_canonical_metric
+                base_point = self.regularize(base_point, point_representation)
+
+                jacobian = self.jacobian_translation(
+                              point=base_point,
+                              left_or_right=metric.left_or_right,
+                              point_representation=point_representation)
+                inv_jacobian = gs.linalg.inv(jacobian)
+                tangent_vec_at_id = gs.dot(
+                        tangent_vec,
+                        gs.transpose(inv_jacobian, axes=(0, 2, 1)))
+                tangent_vec_at_id = gs.squeeze(tangent_vec_at_id, axis=1)
+
+                tangent_vec_at_id = self.regularize_tangent_vec_at_identity(
+                                              tangent_vec_at_id,
+                                              metric,
+                                              point_representation)
+
+                regularized_tangent_vec = gs.dot(tangent_vec_at_id,
+                                                 gs.transpose(jacobian,
+                                                              axes=(0, 2, 1)))
+                regularized_tangent_vec = gs.squeeze(regularized_tangent_vec,
+                                                     axis=1)
+            else:
+                # TODO(nina): is regularization needed in nD?
+                regularized_tangent_vec = tangent_vec
```

Comment:
do you want to make a copy here? because this is not a copy.

Reply:
hm right now I don't know since this 'else' doesn't do much. but let's write gs.copy for later. thanks!